src/components/BigNumber/BigNumber.tsx

```diff
@@ -40,10 +38,25 @@ export function BigNumber({
 		chartView: useRef<VegaView>(),
 		chartWidth: 600,
 	},
+	method
 }: BigNumberProps) {
 	const { chartWidth, height, locale, data, ...rscChartRemain } = rscChartProps;
 
-	const bigNumberValue = data && data.length > 0 ? data[data.length - 1][dataKey] : undefined;
+	if (!method) {
+		method = 'last';
+	}
+
+	let bigNumberValue: number | undefined;
```

Comment:
Could we try refactoring this to make bigNumberValue never undefined?  I'm concerned that having to check for undefined later on makes the code more complex.  It might also make it easier to pass the sonarcloud smells test.  If its too difficult don't worry though

Reply:
Agreed, on line 41 you can do `method = 'last'` and that will set it to that as a default should it be undefined and you don't need to do a check for it, hence why orientation and rscCharProps both also have defaults so I don't have to an undefined check